src/Test.UnitTests.Sarif/TestData/ExpectedOutputs/NoRuleDefaultConfiguration.sarif

```diff
@@ -0,0 +1,55 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+  "version": "2.1.0",
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "FailureLevelRoundtrippingTester",
+          "rules": [
+            {
+              "id": "NoDefaultConfigurationLevel"
+            }
+          ]
+        }
+      },
+      "results": [
+        {
+          "ruleId": "NoDefaultConfigurationLevel/ShouldBeNone",
+          "level": "none",
+          "message": {
+            "text": "Level should be 'none' as it is directly expressed on the result. This is an example of rule trace output."
+          }
+        },
+        {
+          "ruleId": "NoDefaultConfigurationLevel/ShouldBeNone",
+          "kind": "informational",
+          "message": {
+            "text": "Level should be 'none' because kind is not 'fail' (so 'none' is the default)."
+          }
+        },
+        {
+          "ruleId": "NoDefaultConfigurationLevel/ShouldBeNote",
+          "level": "note",
+          "message": {
+            "text": "Level should be 'note' as it is directly expressed on the result."
+          }
+        },
+        {
+          "ruleId": "NoDefaultConfigurationLevel/ShouldBeWarning",
+          "message": {
+            "text": "Level is 'warning' because kind is implictly 'fail' (so 'warning' is the default when there's no default configuration level')."
```

Comment:
>Level is 'warning' because kind is implictly 'fail' [](http://example.com/codeflow?start=21&length=51)

I like how the message is used for information.

Should we also add a case:
Level is 'warning' because kind is **explicit** 'fail'

Reply:
Great question! The answer is no for a strange reason: doing this won't round-trip. :) If you mark a result explicitly as kind == 'fail', that explicit information should be removed on saving the log file (because it is the default value). And so that saved version won't match the input. 

Your test case is a good one, but we should include it elsewhere, i.e., our serialization logic should allow for this value to be explicitly set. Maybe we can then leverage the validation that we'll use here for the round-tripping to make sure everything is still correct.